corporate/lib/stripe.py

```diff
@@ -1205,9 +1205,8 @@ def setup_upgrade_payment_intent_and_charge(
     ) -> str:
         customer = self.update_or_create_stripe_customer()
         assert customer is not None  # for mypy
-        price_per_license = get_price_per_license(
-            plan_tier, billing_schedule, customer.default_discount
-        )
+        discount_for_plan = customer.get_discount_for_plan_tier(plan_tier)
+        price_per_license = get_price_per_license(plan_tier, billing_schedule, discount_for_plan)
```

Comment:
We could also update `price_per_license` and `compute_plan_parameters` to take a customer object as a parameter (instead of the discount decimal) and get the discount as part of what happens in `price_per_license`.

Reply:
Due to the fact that we're likely using this new field for fixed rate plans, and will allow setting it when a billing entity has a current active plan, we won't want to change the behavior for `price_per_license` or `compute_plan_parameters` because for some cases we'll want to use the current plan discount (e.g. changing the billing schedule) and other times we'll want to use the customer default discount (and look at the plan tier).

Reply:
Yeah, that sounds correct to me.